### PR TITLE
[Ameba] Implement Matter OTA header processing

### PR DESCRIPTION
**Pull request by pankore**
#### Problem
* Implement Matter OTA header processing

#### Change overview
* Add ProcessHeader in AmebaOTAImageProcessor.cpp

#### Testing
Tested ota-requestor-app

### CODE REVIEW

**Comment by pankore** [comment]
The prerequisites of this PR are #15252 and #15253.

File: src/platform/Ameba/AmebaOTAImageProcessor.cpp [hunk 1]
```diff
@@ -327,26 +333,63 @@ void AmebaOTAImageProcessor::HandleProcessBlock(intptr_t context)
...
+    if (mHeaderParser.IsInitialized())
+    {
+        OTAImageHeader header;
+        CHIP_ERROR error = mHeaderParser.AccumulateAndDecode(block, header);
+
+        // Needs more data to decode the header
+        ReturnErrorCodeIf(error == CHIP_ERROR_BUFFER_TOO_SMALL, CHIP_NO_ERROR);
+        ReturnErrorOnFailure(error);
+
+        // We save the software version to be used in the next NotifyUpdateApplied, but it's a non-standard
```
Comment (Damian-Nordic) [inline src/platform/Ameba/AmebaOTAImageProcessor.cpp:376]: Are you sure you wanted copy that functionality?

File: src/platform/Ameba/AmebaOTAImageProcessor.cpp [hunk 2]
```diff
@@ -173,13 +176,16 @@ void AmebaOTAImageProcessor::HandleProcessBlock(intptr_t context)
...
+    CHIP_ERROR error = imageProcessor->ProcessHeader(block);
+
 #if defined(CONFIG_PLATFORM_8721D)
     if (!imageProcessor->readHeader) // First block received, process header
     {
         uint8_t * tempBuf          = (uint8_t *) ota_update_malloc(32);
         imageProcessor->pOtaTgtHdr = (update_ota_target_hdr *) ota_update_malloc(sizeof(update_ota_target_hdr));
 
-        memcpy(tempBuf, imageProcessor->mBlock.data(), 32);
+        memcpy(tempBuf, block.data(), 32);
```
Comment (Damian-Nordic) [inline src/platform/Ameba/AmebaOTAImageProcessor.cpp:188]: How do you know block.size() >= 32 after processing the Matter header?

**Comment by pankore** [comment]
The prerequisite for this PR is #15253.

File: src/platform/Ameba/AmebaOTAImageProcessor.cpp (hunk 1 above)
Comment (pankore) [reply src/platform/Ameba/AmebaOTAImageProcessor.cpp]: This has been removed.

File: src/platform/Ameba/AmebaOTAImageProcessor.cpp (hunk 2 above)
Comment (pankore) [reply src/platform/Ameba/AmebaOTAImageProcessor.cpp]: Thanks for pointing this out. This has been fixed.

**Comment by pankore** [comment]
Hi @andy31415 @woody-apple ,
Please kindly help to review this PR, all checks have passed. Thanks!

**Review by andy31415 — APPROVED** [review]

**Comment by pankore** [comment]
Hi [anush-apple](https://github.com/anush-apple), [austinh0](https://github.com/austinh0), [balducci-apple](https://github.com/balducci-apple), [Byungjoo-Lee](https://github.com/Byungjoo-Lee), [bzbarsky-apple](https://github.com/bzbarsky-apple), [carol-apple](https://github.com/carol-apple), [chrisdecenzo](https://github.com/chrisdecenzo), [chshu](https://github.com/chshu), [chulspro](https://github.com/chulspro), [Damian-Nordic](https://github.com/Damian-Nordic), [dhrishi](https://github.com/dhrishi), [electrocucaracha](https://github.com/electrocucaracha), [erjiaqing](https://github.com/erjiaqing), [franck-apple](https://github.com/franck-apple), [gjc13](https://github.com/gjc13), [harimau-qirex](https://github.com/harimau-qirex), [hawk248](https://github.com/hawk248), [holbrookt](https://github.com/holbrookt), [hrajendran-samsung](https://github.com/hrajendran-samsung), [isiu-apple](https://github.com/isiu-apple), [jelderton](https://github.com/jelderton), [jepenven-silabs](https://github.com/jepenven-silabs), [jmartinez-silabs](https://github.com/jmartinez-silabs), [kghost](https://github.com/kghost), [lazarkov](https://github.com/lazarkov), [LuDuda](https://github.com/LuDuda), [lzgrablic02](https://github.com/lzgrablic02) [msandstedt](https://github.com/msandstedt)[robszewczyk](https://github.com/robszewczyk), [sagar-apple](https://github.com/sagar-apple), [saurabhst](https://github.com/saurabhst), [selissia](https://github.com/selissia), [tcarmelveilleux](https://github.com/tcarmelveilleux), [tecimovic](https://github.com/tecimovic), [turon](https://github.com/turon), [vijs](https://github.com/vijs), [vivien-apple](https://github.com/vivien-apple), [wbschiller](https://github.com/wbschiller), [woody-apple](https://github.com/woody-apple)

Please kindly help to check this PR, this auto-tests have all passed. Thanks!

**Pull request merged**